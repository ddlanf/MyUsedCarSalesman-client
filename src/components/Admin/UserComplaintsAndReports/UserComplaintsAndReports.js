import React, { Component } from 'react'

export default class UserComplaintsAndReports extends Component {
    
    renderUserReports(){
        const { reports } = this.props
        const userTable = reports.map(report => {
            return (
                <tr 
                     key={report.id}
                      >
                    <th>{report.id}</th>
                    <th className="admin-paragraph">{report.message}</th>
                    <th>{report.message_type}</th>
                    <th>{report.user_id}</th>
                    <th>{report.date_sent}</th>
                </tr>
            )
        })

        return userTable;
    }
    
    render() {
        return (
            <div className="users-info reports">
                <h2>User Complaints and reports</h2>
                <table className="admin-table reports">
                    <thead>
                        <tr className="about-reports">
                            <th>ID</th>
                            <th>Message</th>
                            <th>Type</th>
                            <th>User ID</th>
                            <th>Date Sent</th>
                        </tr>
                    </thead>
                    <tbody>
                        {this.renderUserReports()}
                    </tbody>
                </table>
            </div>
        )
    }
}
