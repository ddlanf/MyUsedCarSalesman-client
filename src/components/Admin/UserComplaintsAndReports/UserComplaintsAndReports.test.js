import React from 'react';
import ReactDOM from 'react-dom';
import UserComplaintsAndReports from './UserComplaintsAndReports';
import renderer from 'react-test-renderer';

describe('UserComplaintsAndReports component', () => {
    
    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<UserComplaintsAndReports reports={[]}/>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(<UserComplaintsAndReports reports={[]}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});