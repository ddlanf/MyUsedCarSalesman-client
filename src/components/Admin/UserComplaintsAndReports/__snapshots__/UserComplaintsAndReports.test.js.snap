// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserComplaintsAndReports component renders the UI as expected 1`] = `
<div
  className="users-info reports"
>
  <h2>
    User Complaints and reports
  </h2>
  <table
    className="admin-table reports"
  >
    <thead>
      <tr
        className="about-reports"
      >
        <th>
          ID
        </th>
        <th>
          Message
        </th>
        <th>
          Type
        </th>
        <th>
          User ID
        </th>
        <th>
          Date Sent
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
